### PR TITLE
Response status changes

### PR DESCRIPTION
**Pull request by mcombspangea**
This PR will

- Add pangea API exceptions
- Raise those exceptions if requests are not successful

### CODE REVIEW

File: pangea/exceptions.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    """Audit service specific exceptions"""
+
+
+class TreeNotFoundException(AuditException):
+    """Tree was not found during a root inspection"""
+
+
+# Embargo specific exceptions
+class EmbargoException(PangeaException):
+    """Audit service specific exceptions"""
```
Comment (pangea-andrest) [inline pangea/exceptions.py:83]: "Embargo" services...

File: pangea/response.py [hunk 2]
```diff
@@ -31,47 +34,43 @@ class PangeaResponse(object):
     Properties:
         result (obj): "result" field of the API response as documented at:
             [https://docs.dev.pangea.cloud/docs/api/#responses](https://docs.dev.pangea.cloud/docs/api/#responses)
-        status (str): short description message, i.e. "OK"
-        code (int): HTTP status code
+        status (str): Pangea status code
```
Comment (pangea-andrest) [inline pangea/response.py:37]: Maybe sounds better to say "status message"?

File: pangea/response.py [hunk 3]
```diff
@@ -31,47 +34,43 @@ class PangeaResponse(object):
...
-    _status = None
-    _success = False
-
-    def __init__(self, requests_response):
-        self._code = requests_response.status_code
-        self._status = requests_response.reason
-        self._data = JSONObject(requests_response.json())
-        self._success = requests_response.ok
+    def __init__(self, requests_response: requests.Response):
+        data = status = requests_response.json()
```
Comment (pangea-andrest) [inline pangea/response.py:46]: Is it right to init status here and overwrite in next line?

File: pangea/services/audit.py [hunk 4]
```diff
@@ -220,15 +224,19 @@ def handle_log_response(self, response: PangeaResponse, verify: bool, prev_buffe
                 raise Exception(f"Error: Event hash failed.")
```
Comment (pangea-andrest) [inline pangea/services/audit.py:224]: Should we update this exceptions to AuditException?

File: pangea/services/audit.py [hunk 5]
```diff
@@ -220,15 +224,19 @@ def handle_log_response(self, response: PangeaResponse, verify: bool, prev_buffe
...
 
             # verify membership proofs
-            if not verify_membership_proof(node_hash=event_hash, root_hash=new_buffer_root.root_hash, proof=membership_proof):
+            if not verify_membership_proof(
+                node_hash=event_hash, root_hash=new_buffer_root.root_hash, proof=membership_proof
+            ):
                 raise Exception(f"Error: Membership proof failed.")
```
Comment (pangea-andrest) [inline pangea/services/audit.py:230]: Also here and all remaining

File: tests/integration/test_redact.py [hunk 6]
```diff
@@ -33,5 +33,5 @@ def test_redact_structured(self):
     def test_redact_with_structured_data(self):
         data = {"phone": "415-867-5309"}
 
-        response = self.redact.redact(data)
-        self.assertEqual(response.code, 400)
+        with self.assertRaises(exceptions.ValidationException):
```
Comment (pangea-andrest) [inline tests/integration/test_redact.py:36]: Could you add a little comment about what are you testing and why it should raise an exception or not

**Review by pangea-andrest — CHANGES_REQUESTED** [review]

**Comment by pangea-andrest** [comment]
Also it would be great to have at least one test per service raising exception. Could you add them?

File: pangea/response.py (hunk 3 above)
Comment (mcombspangea) [reply pangea/response.py:46]: typo.

File: tests/integration/test_redact.py (hunk 6 above)
Comment (mcombspangea) [reply tests/integration/test_redact.py:36]: This is on line 32

File: pangea/exceptions.py (hunk 1 above)
Comment (pangea-andrest) [reply pangea/exceptions.py:83]: @mcombspangea This one remains, on the comment it says "Audit Service" and it should be "Embargo Service"

**Review by pangea-andrest — APPROVED** [review]

**Pull request merged**